### PR TITLE
Fix hot blob sidecar storage bug

### LINKED ISSUE #6668: Implement unit tests in `DatabaseTest` covering 4844 enabled tx
Write tests in `DatabaseTest`  to cover `KvStoreDatabase::doUpdate` with 4844 flag enabled:
- [ ] blobs confirmation for new hot blocks
- [ ] blobs removal for deleted hot blocks)

related to: #6636

required: #6634

### PR DESCRIPTION
**Pull request by tbenr**
Fixes a bug in which, for storage configurations impling separated hot\finalized DB, we where writing blobs in hot storage instead of the finalized one.

- removes all hot blob sidecar references
- remove unused blobs sidecar storage related methods
- update `DatabaseTest` to always use transactions

fixes #6668

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: storage/src/integration-test/java/tech/pegasys/teku/storage/server/kvstore/DatabaseTest.java [hunk 1]
```diff
@@ -2214,45 +2207,62 @@ private void assertBlobSidecarsAvailabilityExceptPruned(
...
+              .collect(Collectors.toUnmodifiableList()),
+          spec.calculateBlockCheckpoints(block.getState()));
     }
     commit(transaction);
   }
 
-  private void addBlobSidecars(final List<BlobSidecar> blobSidecars) {
-    blobSidecars.forEach(blobSidecar -> database.storeBlobSidecar(blobSidecar));
+  private void add(final Collection<SignedBlockAndState> blocks) {
+    final StoreTransaction transaction = recentChainData.startStoreTransaction();
```
Comment (zilm13) [inline storage/src/integration-test/java/tech/pegasys/teku/storage/server/kvstore/DatabaseTest.java]: Maybe one line `add(blocks, List.of());` instead?

File: storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java [hunk 2]
```diff
@@ -950,23 +947,42 @@ private UpdateResult doUpdate(final StorageUpdate update) {
       LOG.trace("Committing hot db changes");
       updater.commit();
     }
+
+    if (update.isBlobSidecarsEnabled()) {
+      updateBlobSidecarData(
```
Comment (zilm13) [inline storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java]: Maybe move this a screen above to the previous  `if (update.isBlobSidecarsEnabled()) {` block?

**Review by zilm13 — APPROVED** [review]
LGTM

File: storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java (hunk 2 above)
Comment (tbenr) [reply storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java]: yes... in the current form, it is also much safer to write blobs before blocks.

File: storage/src/integration-test/java/tech/pegasys/teku/storage/server/kvstore/DatabaseTest.java (hunk 1 above)
Comment (tbenr) [reply storage/src/integration-test/java/tech/pegasys/teku/storage/server/kvstore/DatabaseTest.java]: no strong opinion on that. OK

Comment (tbenr) [reply storage/src/integration-test/java/tech/pegasys/teku/storage/server/kvstore/DatabaseTest.java]: ah sorry, now I understand :)

**Pull request merged**